src/daft-context/src/lib.rs

```diff
@@ -328,24 +360,29 @@ fn get_runner_config_from_env() -> RunnerConfig {
         Some(())
     });
 
-    match runner_from_envvar.to_lowercase().as_str() {
-        "ray" => RunnerConfig::Ray {
-            address,
-            max_task_backlog,
-            force_client_mode,
-        },
-        "py" => RunnerConfig::Py {
-            use_thread_pool: std::env::var(DAFT_DEVELOPER_USE_THREAD_POOL)
-                .ok()
-                .map(|s| matches!(s.trim().to_lowercase().as_str(), "true" | "1")),
-            num_threads: None,
-        },
-        _ if !in_ray_worker && (ray_is_initialized || ray_is_in_job) => RunnerConfig::Ray {
-            address: None,
-            max_task_backlog,
-            force_client_mode,
-        },
-        _ => RunnerConfig::Native { num_threads: None },
+    (ray_is_in_job, in_ray_worker, ray_is_initialized)
+}
+
+#[cfg(feature = "python")]
+fn get_runner_config_from_env() -> RunnerConfig {
+    const DAFT_RUNNER: &str = "DAFT_RUNNER";
+
+    let runner_from_envvar = std::env::var(DAFT_RUNNER)
+        .unwrap_or_default()
+        .to_lowercase();
+
+    match runner_from_envvar.as_str() {
+        "native" => RunnerConfig::Native { num_threads: None },
+        "ray" => get_ray_runner_config_from_env(true),
+        "py" => get_py_runner_config_from_env(),
+        _ => {
+            let (ray_is_in_job, in_ray_worker, ray_is_initialized) = detect_ray_state();
+            if !in_ray_worker && (ray_is_initialized || ray_is_in_job) {
```

Comment:
```suggestion
            let ray_detected = detect_ray_state();
            if ray_detected {
```

and then u abstract the boolean nonsense in `detect_ray_state`

Reply:
and maybe we can do 
```
"ray" | _  if detect_ray_state() => get_ray_runner_config_from_env
_ => RunnerConfig::native
```